src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -287,13 +300,15 @@ private Command BuildDeleteCommand()
 
         private Command BuildListCommand()
         {
-            var listCommand = new Command("list-deployments", "List existing deployments.")
+            var listCommand = new Command("list-deployments", "List existing deployments.");
+            lock (s_locker)
             {
-                _optionProfile,
-                _optionRegion,
-                _optionProjectPath,
-                _optionDiagnosticLogging
-            };
+                listCommand.AddOption(_optionProfile);
+                listCommand.AddOption(_optionRegion);
```

Comment:
same comment about consistency

Reply:
This is probably done to achieve thread safety. @ganeshnj can weigh more on this.

Reply:
https://github.com/aws/aws-dotnet-deploy/pull/293/files#r688018748

Reply:
Fixed